### DISCUSSION TITLE
Does Prisma not support one to one relation with PK refer to PK?

### DISCUSSION THREAD

**Discussion by tetthys** [prisma/prisma#23276] [Q&A] [answered]
### Question

I added some one to one relations.

I don't want this structure.

I want :

User
  id -> Card.id

Card
  id

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @tetthys 👋 

Thank you for raising this question.

You can create a schema that looks like this
```prisma
model User {
  id   Int    @id
  name String @db.VarChar(255)
  card Card?
}

model Card {
  id   Int  @id
  user User @relation(fields: [id], references: [id])
}
```

However, it's important to note that this approach of directly referencing the same primary key for a one-to-one relationship is not recommended due to:

- Redundancy: The user ID is stored in both tables, increasing storage usage and potentially leading to inconsistencies.
- Foreign Key Constraint Violation: When creating a new card, its ID must already exist in the user table, potentially creating constraint violations.

You should always use the recommended approach for creating one-to-one relations in Prisma as described in the [documentation](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/one-to-one-relations)

**Reply by tetthys** [reply]
Thank you!
